# HG changeset patch
# User Fabian Groffen <grobian@gentoo.org>
# Date 1336718933 -7200
# Branch 2.7
# Node ID d71c265f518345c30b83e87a49545147c9375979
# Parent  f8bc2f32d80b5a9459e829947dad56839d8d15de
http://bugs.python.org/issue10898
https://bugs.gentoo.org/show_bug.cgi?id=351462

upstream is trying to solve this rather clumpsy, let's hope they'll get
it right at some point

diff -r f8bc2f32d80b -r d71c265f5183 Modules/posixmodule.c
--- a/Modules/posixmodule.c	Fri May 11 08:48:52 2012 +0200
+++ b/Modules/posixmodule.c	Fri May 11 08:48:53 2012 +0200
@@ -838,11 +838,11 @@
 #undef STRUCT_STAT
 #if defined(MS_WIN64) || defined(MS_WINDOWS)
 #       define STAT win32_stat
-#       define FSTAT win32_fstat
+#       define PYTHON_FSTAT win32_fstat
 #       define STRUCT_STAT struct win32_stat
 #else
 #       define STAT stat
-#       define FSTAT fstat
+#       define PYTHON_FSTAT fstat
 #       define STRUCT_STAT struct stat
 #endif
 
@@ -6696,7 +6696,7 @@
     if (!_PyVerify_fd(fd))
         return posix_error();
     Py_BEGIN_ALLOW_THREADS
-    res = FSTAT(fd, &st);
+    res = PYTHON_FSTAT(fd, &st);
     Py_END_ALLOW_THREADS
     if (res != 0) {
 #ifdef MS_WINDOWS
